src/concepts/segments.md

```diff
@@ -0,0 +1,134 @@
+
+This is a set of three segments. 
+On a given day, every client falls into exactly one of these segments.
+Each client's segment is stored in `telemetry.clients_last_seen.segment_regular_users_v1`.
+(FIXME: or it will be if the [PR](https://github.com/mozilla/bigquery-etl/pull/825) is accepted in a form similar to its current state)
+
+Regular users v1 is defined as 
+clients who have at least two days where they browsed >=5 URIs, 
+in *all four* of the last weeks. 
+Observationally, on any given day this segment seems to contain a large fraction of our users 
+and has exceptionally high retention.
+
+Semi-regular users v1 is defined as 
+clients who browsed >=5 URIs on at least two days in each of *one, two or three* of the last four weeks. 
+This seems to be a smaller segment of our users, and has relatively high retention.
+
+New/irregular users v1 is defined as 
+clients who browsed >=5 URIS on at least two days in *none* of the last four weeks. 
+This is a smaller segment of our users and has low retention 
+(though "activation" is likely a more relevant word than "retention" for many of these clients).
+
+In each of these definitions, 
+"the last four weeks" refer to 7 day periods that end on the day of week 
+before the first day for which we want to study the user. 
+They do not necessarily run Monday through Sunday.
+
+## Obsolete segments
+
+None yet.
+
+## Writing queries with segments
+
+When a segment is defined with respect to a user's _behavior_ (e.g. usage levels) 
+as opposed to more stable traits (e.g. country), 
+we should evaluate each user's segment eligibility 
+using data collected _before_ the time period in which we want to study their actions. 
+Else we run the risk of making trivial discoveries 
+like "heavy users use the product heavily" instead of more meaningful ones 
+like "heavy users go on to continue using the product heavily".
+
+So, when writing queries, 
+be sure to compute users' segments using only 
+data collected before the time period you're analysing their behavior. 
+For example, we should compute DAU for the "regular users v1" segment on 2020-03-18 
+by checking the usage criterion with data up to and including 2020-03-17.
+
+Segments are found as columns in the `clients_last_seen` dataset: the segment listed for a client on a `submission_date` is valid for that `submission_date` because it is computed only using behavioral data collected _before_ the `submission_date`.
+
+### WAU and MAU
+
+Users can change segments part way through a week or a month. 
+When computing MAU for a segment, 
+include all clients that fit that segment for at least one day they were active in the month, 
+regardless of whether they fit a contradictory segment on another active day. 
+This means that the sum of MAU for each non-overlapping segment can be larger than MAU.
+(FIXME: This point needs special attention in the review process - it appears to be the best option)
```

Comment:
I'd disagree.  I think that for any given metricXprofileXdate, there should be a single segment.  We can discuss elsewhere maybe.

Reply:
OK, if we're going to have this debate then to ground it I think I should spend a day writing up a doc that outlines the plausible options, and the pros and cons of each of them. I'll consider usefulness to decision making, ease of interpretation, ease of implementation, and surprise in the pros and cons. As far as I'm aware, there is no clear right answer here - each option involves some ickiness and/or surprise - so if your instincts differ to mine then we should check this out thoroughly.

Reply:
I guess I wonder why I want to know "how many users ever had each label in the last month" and not "of the users we saw in the last month, how many have this label today"? The latter seems more responsive; zero-sum changes due to transitions seem easier to interpret than an inflation of one group followed by a deflation of another.

If these are states that a user can move through, it doesn't really make sense to have a user in a superposition of states. But maybe a state machine is a bad model and maybe that's not what we're plotting.

Which is to say I'll also read this doc with interest!

Reply:
[Here's the doc](https://docs.google.com/document/d/1OpejuD-wtViUX4nwddmaIHF85hU8WknjJBT-zjxxgkE/) - it's probably best to have the discussion in it (inline comments and/or writing "X's take" sections at the end).